pingcap/tiflow
# etcd (ticdc): fix etcd transaction operation out of limit.

### What problem does this PR solve?


Issue Number: close #7131 

### What is changed and how it works?
Add `taskPosition`(s) cleaning logic to `func  DeleteCaptureInfo()` which will be call in a `defer` block when the capture exits, so all `taskPosition`(s) would be cleaned up at the same time. 

If we don't do this, when the `processor` or `owner` panic in the second `tick`, the stale  `taskPosition`(s) would leak since the `etcdWorker` can nerver go up to the stale keys cleaning logic. 

Finally, there will be a huge amount `taskPosition`(s) store in etcd and lead to a `etcd transaction operation out of limit` error.

### Check List 

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
1. Add a `log.Panic()` below `processor.go` line 559,  make cdc, and start the cdc server again and again.
2. Use `etcdctl` to check is there are leaked `taskPosition`.

#### Questions 

##### Will it cause performance regression or break compatibility?

##### Do you need to update user documentation, design documentation or monitoring documentation?

### Release note 

```release-note
Fix a bug that may cause cdc unavaliable by commit too many operation in a etcd transaction.
```
